acnportal/acnsim/base.py

```diff
@@ -0,0 +1,473 @@
+        }
+        if dependency_versions is not None:
+            for pkg in dependency_versions.keys():
+                if (current_dependency_versions[pkg]
+                        != dependency_versions[pkg]):
+                    warnings.warn(
+                        f"Current version of dependency {pkg} does not "
+                        f"match serialized version. "
+                        f"Current: {current_dependency_versions[pkg]}, "
+                        f"Serialized: {dependency_versions[pkg]}.",
+                    )
+
+        try:
+            obj_dict = context_dict[obj_id]
+        except KeyError:
+            raise KeyError(
+                f"Object with ID {obj_id} not found in context_dict.")
+
+        # Check if this object has already been converted, and return
+        # the appropriate dict if this is the case.
+        if obj_id is not None and obj_id in loaded_dict:
+            return loaded_dict[obj_id], loaded_dict
+
+        if obj_dict['class'] != f'{cls.__module__}.{cls.__name__}':
+            warnings.warn(
+                f"Deserializing as type "
+                f"{cls.__module__}.{cls.__name__}. Object was "
+                f"serialized as type {obj_dict['class']}.",
+                UserWarning
+            )
+
+        # attribute_dict is a dict mapping attribute names to JSON
+        # serializable values.
+        attribute_dict = obj_dict['attributes']
+
+        # Call this class' from_dict method to convert the JSON
+        # representation of this object's attributes into the actual
+        # object.
+        out_obj, loaded_dict = cls.from_dict(
+            attribute_dict, context_dict, loaded_dict, cls_kwargs)
+
+        # Check that all attributes have been loaded.
+        # Warn if some attributes weren't loaded.
+        if out_obj.__dict__.keys() != attribute_dict.keys():
+            unloaded_attrs = (set(attribute_dict.keys())
+                              - set(out_obj.__dict__.keys()))
+            warnings.warn(
+                f"Attributes {unloaded_attrs} present in object of "
+                f"type {obj_dict['class']} but not handled by object's "
+                f"from_dict method. Loaded object may have inaccurate "
+                f"attributes.",
+                UserWarning
+            )
+            for attr in unloaded_attrs:
+                # Try reading this attribute from an ID in
+                # attribute_dict.
+                try:
+                    out_attr, loaded_dict = build_from_id(
+                        attribute_dict[attr],
+                        context_dict,
+                        loaded_dict=loaded_dict
+                    )
+                    setattr(out_obj, attr, out_attr)
+                except (KeyError, TypeError):
+                    warnings.warn(
+                        f"Loader for attribute {attr} not found. "
+                        f"Setting attribute {attr} directly.",
```

Comment:
Instead of setting attributes to the repr string, set it to a dummy object that throws an error whenever it's called. Override getattr.